hanish/darksky.py

```diff
@@ -0,0 +1,192 @@
+# hanish.darksky
+# Handlers and functionality for making Dark Sky API requests.
+#
+# Author:   Benjamin Bengfort <benjamin@bengfort.com>
+# Created:  Mon May 08 12:04:11 2017 -0400
+#
+# Copyright (C) 2016 Bengfort.com
+# For license information, see LICENSE.txt
+#
+# ID: cdarksky.py [] benjamin@bengfort.com $
+
+"""
+Handlers and functionality for making Dark Sky API requests.
+"""
+
+##########################################################################
+## Imports and Module Vars
+##########################################################################
+
+import requests
+
+try:
+    # Python 3
+    from urllib.parse import urljoin
+except ImportError:
+    # Python 2
+    from urlparse import urljoin
+
+
+from datetime import date, datetime
+from expiringdict import ExpiringDict
+from .exceptions import HanishValueError
+from .exceptions import DarkSkyException
+
+
+DARKSKY_API_URL  = "https://api.darksky.net"
+API_CALLS_HEADER = "X-Forecast-API-Calls"
+
+
+##########################################################################
+## DarkSky API
+##########################################################################
+
+class DarkSky(object):
```

Comment:
The `cache` and `limit` features are nice additions. A question that sort of relates to both of them:

If we wanted to have the option for various caching backends, what would our options be and which might you choose?